.github/workflows/tests@v1.yml

```diff
@@ -0,0 +1,221 @@
+
+  verify:
+    name: Full verify
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        java-version: [8, 11]
+      fail-fast: false
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Set up JDK ${{ matrix.java-version }}
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java-version }}
+          distribution: 'adopt'
+
+      - name: Full verify
+        run: mvn -B verify -DskipTests
+
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+
+      - name: Run unit tests
+        run: mvn -B test -Dfmt.skip=true -Dclirr.skip=true -Danimal.sniffer.skip=true
+
+      - name: Copy test results
+        if: success() || failure()
+        run: |
+          shopt -s globstar
+          mkdir unit
+          cp --parents ./**/target/*-reports/*.xml unit/
+
+      - name: Upload test results
+        uses: actions/upload-artifact@v2
+        if: success() || failure()
+        with:
+          name: test-results
+          path: "*/**/target/*-reports/*.xml"
+
+  setup-integration-tests:
+    name: Setup ITs
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Setup Python 3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+
+      - name: Fetch Scylla and Cassandra versions
+        id: fetch-versions
+        run: |
+          pip3 install -r ci/requirements.txt
+          echo "::set-output name=scylla-integration-tests-versions::$(python3 ci/version_fetch.py scylla-oss-stable:2 scylla-oss-rc scylla-enterprise-stable:2 scylla-enterprise-rc)"
+          echo "::set-output name=cassandra-integration-tests-versions::$(python3 ci/version_fetch.py cassandra3-stable:1)"
+
+    outputs:
+      scylla-integration-tests-versions: ${{ steps.fetch-versions.outputs.scylla-integration-tests-versions }}
+      cassandra-integration-tests-versions: ${{ steps.fetch-versions.outputs.cassandra-integration-tests-versions }}
+
+  cassandra-integration-tests:
+    name: Cassandra ITs
+    runs-on: ubuntu-latest
+    needs: [setup-integration-tests]
+    timeout-minutes: 90
+
+    strategy:
+      matrix:
+        cassandra-version: ${{ fromJson(needs.setup-integration-tests.outputs.cassandra-integration-tests-versions) }}
+      fail-fast: false
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
```

Comment:
In `actions/setup-java@v2` there is support for caching Maven dependencies, but decided not to enable this to verify that build from scratch works.